### PR TITLE
Refactor/patch object on save

### PR DESCRIPTION
**Pull request by genie9**
### Description

Move responsibility of adding an object to a folder on object creation and update from frontend to object endpoint. 

###Related issues
Closes https://github.com/CSCfi/metadata-submitter/issues/335

### Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
Frontend now should add folder ID as a query parameter with object POST.

### Changes Made
Folder id is now required query param with object POST.
A folder is patched after an object is created and updated.
Adds extraction of title for folder's objects display title.

### Testing
- [x] Integration Tests

### CODE REVIEW

**Comment by blankdots** [comment]
how is this different than #340 ?

**Comment by blankdots** [comment]
Requires https://github.com/CSCfi/metadata-submitter-frontend/issues/626 in front-end

File: metadata_backend/api/handlers/object.py [hunk 1]
```diff
@@ -283,3 +305,43 @@ async def patch_object(self, req: Request) -> Response:
...
+
+        :param schema: schema of objects to be added to the folder
+        :param ids: object IDs
+        :param params: addidtional data required for db entry
+        :returns: list of patch operations
+        """
+        if not params.get("cont_type", None):
+            submission_type = "Form"
+        else:
+            submission_type = params["cont_type"].upper()
```
Comment (blankdots) [inline metadata_backend/api/handlers/object.py:320]: realised that for this to work we need to add `CSV` to lines:
- https://github.com/CSCfi/metadata-submitter/blob/develop/metadata_backend/helpers/schemas/folders.json#L999
- https://github.com/CSCfi/metadata-submitter/blob/develop/metadata_backend/helpers/schemas/folders.json#L1037

**Review by blankdots — COMMENTED** [review]
if we add the object in the folder when creating, should we still allow patching https://github.com/CSCfi/metadata-submitter/blob/develop/metadata_backend/api/handlers/folder.py#L38 `metadataObjects` and `drafts` when PATCH folder ?

File: metadata_backend/api/handlers/object.py (hunk 1 above)
Comment (genie9) [reply metadata_backend/api/handlers/object.py:320]: strangely it worked without schema changes.

**Comment by genie9** [comment]
> how is this different than #340 ?

changed the branch name because realized that we are not patching on the update. This obviously closed the PR.

**Comment by genie9** [comment]
> if we add the object in the folder when creating, should we still allow patching https://github.com/CSCfi/metadata-submitter/blob/develop/metadata_backend/api/handlers/folder.py#L38 `metadataObjects` and `drafts` when PATCH folder ?

If e.g. title of an object will change i think it should be updated to the folder `metadataObjects` also.

**Comment by blankdots** [comment]
> If e.g. title of an object will change i think it should be updated to the folder `metadataObjects` and `drafts` also. And actualy this is not happening now.

Good point, I think front-end does a PATCH or PUT for that I don't remember we need to check.

**Review by blankdots — COMMENTED** [review]
looks good.
please add changes to `CHANGELOG.md`

**Review by blankdots — APPROVED** [review]

**Pull request merged**